meta/og header tag fixes

michaelmoussa opened the pull request:
See internal JIRA ticket `INFODEV-2967` for further details and motivation behind this PR.

**Summary of changes:**
* Add post title as a fallback for ogTitle
* Add Rackspace Fanatiguy as fallback for ogImage on blog posts

**NOTE:** As was the case with my other Pull Request, #955, I was not able to test this locally.

michaelmoussa committed:
meta/og fixes

* Add post title as a fallback for ogTitle
* Add Rackspace Fanatiguy as fallback for ogImage on blog posts

michaelmoussa committed:
Use blog post excerpt as Description if none provided

robb-romans commented:
good stuff. just a quick note that we should avoid adding more of those images with the Fanatiguy, since we are in the process of an overall docs site redesign and all the icons may change soon.

michaelmoussa commented:
@robb-romans Should I remove it entirely, then? Or use a different one instead? Merging this as-is would end up resulting in all newly shared blog posts having the Fanatiguy preview image instead of [this one](https://657cea1304d5d92ee105-33ee89321dddef28209b83f19f06774f.ssl.cf1.rackcdn.com/default-og-image-46fb3587dedfdf950188fabbddd596d67e6b699374a7f4e36b43046d7a24fd09.jpg).

michaelmoussa committed:
Fix excerpt usage when no description present

The `.excerpt` is not available in the envelope, so instead, use the
first 160 characters of the body text.

michaelmoussa commented:
I resolved the local testing issues I was having earlier and verified the changes work as expected after some additional tweaks.

In cases where the og/meta/twitter/etc information is specified in Front Matter of a post, this is an example result:

```html
<title>Lambda Custom Runtimes: A Practical ExamPHPle</title>
  <meta name="description" content="Learn how to build an example PHP runtime using the new AWS Lambda custom runtime API and layering capabilities.">
  <meta name="og:title" content="Lambda Custom Runtimes: A Practical ExamPHPle">
  <meta name="og:description" content="Learn how to build an example PHP runtime using the new AWS Lambda custom runtime API and layering capabilities.">
  <meta name="og:image" content="/assets/src/img/default-blog-post-og-image.png">
  <meta name="og:type" content="site">
  <meta name="twitter:title" content="Lambda Custom Runtimes: A Practical ExamPHPle">
  <meta name="twitter:description" content="Learn how to build an example PHP runtime using the new AWS Lambda custom runtime API and layering capabilities.">
  <meta name="twitter:creator" content="@michaelmoussa">
  <meta name="twitter:site" content="@rackspace">
```

In cases where the information _isn't_ specified, it falls back on the title and part of the body of the post:

```html

<title>Upgrade NetScaler Application Delivery Controller</title>
  <meta name="description" content="NetScaler&reg; Application Delivery Controller (ADC), Citrix&reg; Systems&#39; core networking product, is a tool that improves the delivery speed and quality...">
  <meta name="og:title" content="Upgrade NetScaler Application Delivery Controller">
  <meta name="og:description" content="NetScaler&reg; Application Delivery Controller (ADC), Citrix&reg; Systems&#39; core networking product, is a tool that improves the delivery speed and quality...">
  <meta name="og:image" content="/assets/src/img/default-blog-post-og-image.png">
  <meta name="og:type" content="site">
  <meta name="twitter:title" content="Rackspace Developer Center">
  <meta name="twitter:description" content="NetScaler&reg; Application Delivery Controller (ADC), Citrix&reg; Systems&#39; core networking product, is a tool that improves the delivery speed and quality...">
  <meta name="twitter:creator" content="@rackspace">
```

Previously, it would use the title/description of the parent blog index, which would be less relevant in this case.

This should be safe to merge.

svetkars commented:
Tested locally and compared to the published pages. [This article](https://developer.rackspace.com/blog/Introduction-to-load-and-performance-testing-with-gatling-tool/) shows the generic description at developer.rackspace.com while the local build shows the description of the article.

At developer.rackspace.com:

```<meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>Introduction to Load and Performance Testing with Gatling Tool</title>
        <meta name="description" content="The Rackspace Developer Center gives developers tools and resources to build powerful, scalable apps on the only cloud backed by Fanatical Support®">
        <meta name="og:title" content="Rackspace Developer Center">
        <meta name="og:description" content="The Rackspace Developer Center gives developers tools and resources to build powerful, scalable apps on the only cloud backed by Fanatical Support®">
        <meta name="og:image" content="https://657cea1304d5d92ee105-33ee89321dddef28209b83f19f06774f.ssl.cf1.rackcdn.com/default-og-image-46fb3587dedfdf950188fabbddd596d67e6b699374a7f4e36b43046d7a24fd09.jpg">
        <meta name="og:type" content="site">
        <meta name="twitter:title" content="Rackspace Developer Center">
        <meta name="twitter:description" content="The Rackspace Developer Center gives developers tools and resources to build powerful, scalable apps on the only cloud backed by Fanatical Support®">
        <meta name="twitter:creator" content="@rackspace">
        <meta name="twitter:site" content="@rackspace">
```

In the local build:

```<meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>Introduction to Load and Performance Testing with Gatling Tool</title>
        <meta name="description" content="This blog explores the fundamentals of Load and Performance Testing as well as the basics of the Gatling Tool (https://gatling.io). Some popular testing tools...">
        <meta name="og:title" content="Introduction to Load and Performance Testing with Gatling Tool">
        <meta name="og:description" content="This blog explores the fundamentals of Load and Performance Testing as well as the basics of the Gatling Tool (https://gatling.io). Some popular testing tools...">
        <meta name="og:image" content="/assets/src/img/default-blog-post-og-image.png">
        <meta name="og:type" content="site">
        <meta name="twitter:title" content="Rackspace Developer Center">
        <meta name="twitter:description" content="This blog explores the fundamentals of Load and Performance Testing as well as the basics of the Gatling Tool (https://gatling.io). Some popular testing tools...">
        <meta name="twitter:creator" content="@rackspace">
        <meta name="twitter:site" content="@rackspace">
```

This PR looks good to me. The correct image display functionality will be added in a separate PR.
